pkg/devfile/image/docker_compatible.go

```diff
@@ -24,22 +25,39 @@ func NewDockerCompatibleBackend(name string) *DockerCompatibleBackend {
 
 // Build an image, as defined in devfile, using a Docker compatible CLI
 func (o *DockerCompatibleBackend) Build(image *devfile.ImageComponent, devfilePath string) error {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+
 	if strings.HasPrefix(image.Dockerfile.Uri, "http") {
 		return errors.New("HTTP URL for uri is not supported")
 	}
 
-	log.Infof("\nBuilding image %s", image.ImageName)
+	buildSpinner := log.Spinner("Building image locally")
+	defer buildSpinner.End(false)
 
 	shell := getShellCommand(o.name, image, devfilePath)
 
 	cmd := exec.Command("bash", "-c", shell)
 	cmd.Env = append(os.Environ(), "PROJECTS_ROOT="+devfilePath)
-	cmd.Stdout = os.Stdout
-	cmd.Stderr = os.Stderr
+
+	// If we are not in debug mode, hide the output
+	if log.IsDebug() {
+		cmd.Stdout = log.GetStdout()
+		cmd.Stderr = log.GetStderr()
+	} else {
+		cmd.Stdout = &stdout
+		cmd.Stderr = &stderr
+	}
+
 	err := cmd.Run()
 	if err != nil {
-		return fmt.Errorf("error running %s command: %w", o.name, err)
+		if log.IsDebug() {
+			return fmt.Errorf("error running %s command: %w", o.name, err)
+		}
+		return fmt.Errorf("error running %s command: %w\n\n --- OUTPUT --- \n%s\n\n --- ERROR OUTPUT --- \n%s", o.name, err, cmd.Stdout, cmd.Stderr)
```

Comment:
Can you only print output/error output if they are not empty?
```sh
➜  odo deploy
  __
 /  \__     Deploying the Devfile: nodejs-prj1-api-abhz
 \__/  \    Namespace: default
 /  \__/    CLI Version: v2.5.0
 \__/

↪ Building & Pushing Container: quay.io/unknown-account/myimage
 ✓  Building image locally [1s]
 ✗  Pushing image to container registry [5s]
 ✗  error running podman command: exit status 125

 --- OUTPUT --- 


 --- ERROR OUTPUT --- 
Getting image source signatures
Error: trying to reuse blob sha256:62a747bf1719d2d37fff5670ed40de6900a95743172de1b4434cb019b56f30b4 at destination: checking whether a blob sha256:62a747bf1719d2d37fff5670ed40de6900a95743172de1b4434cb019b56f30b4 exists in quay.io/unknown-account/myimage: unauthorized: authentication required

➜  export PODMAN_CMD=echo
```
The above empty OUTPUT seems a little weird to me.